Source/Core/Common/WorkQueueThread.h

```diff
@@ -70,6 +90,19 @@ class WorkQueueThread
     }
   }
 
+  // Doesn't return until the most recent function invocation has finished.
+  void Flush()
+  {
+    if (m_thread.joinable())
+    {
+      m_flush.Set();
+      Clear();
```

Comment:
Remove this `Clear()` Replace it with a `m_wakeup.Set();`

Otherwise it will drop submissions

Reply:
IMO having 3 different flags and 2 events here makes this a lot more complicated than if we just used one or two condition variables directly.

Reply:
Event is just a nicely wrapped condition var. Flag is just a fancy atomic bool. 

I don't think you simplify things much by using condition vars directly. You still need two of them (one for each direction), and you still need two or three atomic bools. 

Remember, for my usecase, I need to be able to directly test m_flush from with my work function, which might otherwise loop forever.